modules/directives/src/main/scala/scala/build/directives/DirectiveSpecialSyntax.scala

```diff
@@ -0,0 +1,28 @@
+package scala.build.directives
+
+import scala.util.matching.Regex
+
+object DirectiveSpecialSyntax {
+
+  /** Replaces the `${.}` pattern in the directive value with the parent directory of the file
+    * containing the directive. Skips replacement if the pattern is preceded by two dollar signs
+    * ($$). https://github.com/VirtusLab/scala-cli/issues/1098
+    *
+    * @param directiveValue
+    *   the value of the directive, e.g., "-coverage-out:${.}" for example for the directive "//>
+    *   using options "-coverage-out:${.}""
+    * @param path
+    *   the file path from which the directive is read; replacement occurs only if the directive is
+    *   from a local file
+    * @return
+    *   the directive value with the `${.}` pattern replaced by the parent directory, if applicable
+    */
+  def handlingSpecialPathSyntax(directiveValue: String, path: Either[String, os.Path]): String = {
+    val pattern: Regex = """(?<!\$)\$\{\.}""".r
```

Comment:
That won't work with `$$${.}`, which should actually replace the trailing `${.}` (and prepend it with a *single* `$`).

Reply:
Thanks for review @sjrd!